### ISSUE TITLE
Automatic code hinting, without hitting tab

### ISSUE THREAD

**Issue by rkpatel33** [jupyter/notebook#1289]
Is it possible to turn on autocomplete so that the suggestion list is displayed on every letter key press in edit mode?

This would bring the typing experience closer to that of Atom or Sublime code complete and would make switching back and forth just a bit easier.

Would love to see this if it does not exist.

**Comment by captainsafia** [comment]
Thanks for opening this issue, @rkpatel33! I think this is a great idea. If it can be done, it would probably have to be done as an extension that queries the backend with a `complete_request` message every X milliseconds for a completion information. I'll look into this and report back.

**Comment by rkpatel33** [comment]
Thanks, great! Maybe I am missing something, but couldn't it be done by
simply tying whatever event is triggered on <tab> keypress (in edit mode)
to keypresses of [A-Za-z]?


______________________________________
Rishi K. Patel
​, ​
Founder
​,
www.docnavapp.com
*​*
​
rishi@docnavlab.com
​,
917-647-0892​
LinkedIn <https://www.linkedin.com/in/rkpatel33>, @docnavlab
<https://twitter.com/docnavlab>

**Comment by captainsafia** [comment]
I think that would cause a lot of latency on every keypress which would make the experience not as effective.

**Comment by rkpatel33** [comment]
Yeah, I suppose it could. Ok thanks, let me know if you figure anything out
on it.

___________________
Rishi K. Patel
rishi@docnavlab com
917-647-0892

**Comment by janschulz** [comment]
Maybe report this issue in the [ipython-notebook-extensions repo](https://github.com/ipython-contrib/IPython-notebook-extensions/issues), there such requests are usually met with some idea how to implement it with an extension :-)

**Comment by rkpatel33** [comment]
Will do, thanks.

**Comment by ellisonbg** [comment]
In JupyterLab we are going to be redesigning the tooltip and tab completion
UI - will definitely be thinking about this type of thing in that redesign.

On Sat, Apr 2, 2016 at 11:37 AM, rkpatel33 <notifications@github.com> wrote:

> Will do, thanks.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/issues/1289#issuecomment-204770984>
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by rkpatel33** [comment]
Great, there is an extension first cut already working in the extensions repo:

https://github.com/ipython-contrib/IPython-notebook-extensions/pull/564

I added making the dropdown box filtering fuzzy as well, helps a lot.

**Comment by pohmelie** [comment]
Voting up. This is extremely useful feature.

**Comment by baijifeilong** [comment]
On auto completion

`bpython` > `ptpython` > `ipython`

**Comment by bobmayuze** [comment]
Or try this extension: https://github.com/jupyterlab/jupyterlab-monaco

**Comment by GF-Huang** [comment]
what's the progress?

**Comment by kevin-bates** [comment]
Hi @GF-Huang - unless the extension referenced in https://github.com/jupyter/notebook/issues/1289#issuecomment-205422381 above addresses this issue, I'd say there hasn't been any progress given the timestamps on the latest comments. 

If that doesn't address things in an appropriate manner and given the state of this repository, you're probably best off opening this issue in [jupyterlab](https://github.com/jupyterlab/jupyterlab).
